### PR TITLE
feat(profiling): add table sorting

### PR DESCRIPTION
**Pull request by JonasBa**
Allow users to sort the table in asc/desc order. Note that this sorts at node level, so it preserves the hierarchy.

CleanShot 2022-05-13 at 10 52 11

### CODE REVIEW

File: static/app/components/profiling/frameStack.tsx [hunk 1]
```diff
@@ -34,36 +35,125 @@ function formatColorForFrame(
...
     <FrameBar>
       <FrameCallersTable>
         <FrameCallersTableHeader>
           <tr>
-            <th>{t('Self Time')}</th>
-            <th>{t('Total Time')}</th>
-            <th colSpan={100}>{t('Frame')}</th>
+            <th>
+              <TableHeaderButton onClick={() => onSortChange('self weight')}>
+                {t('Self Time')}{' '}
```
Comment (Zylphrex) [inline static/app/components/profiling/frameStack.tsx:120]: nit:
```suggestion
                {t('Self Time ')}
```

File: static/app/components/profiling/frameStack.tsx [hunk 2]
```diff
@@ -34,36 +35,125 @@ function formatColorForFrame(
...
 
-function FrameCallTreeStack({flamegraphRenderer, roots}: FrameCallTreeStackProps) {
+function FrameCallTreeStack({
+  flamegraphRenderer,
+  roots,
+  referenceNode,
+}: FrameCallTreeStackProps) {
+  const [sort, setSort] = useState<'total weight' | 'self weight' | 'name' | null>(
+    'total weight'
+  );
```
Comment (Zylphrex) [inline static/app/components/profiling/frameStack.tsx:88]: Why does this need `null`?

**Review by Zylphrex — APPROVED** [review]

File: static/app/components/profiling/frameStack.tsx (hunk 2 above)
Comment (Zylphrex) [reply static/app/components/profiling/frameStack.tsx:88]: Is the default here not `total weight`? I don't see any place that will set this to `null`

**Review by Jesse-Box — APPROVED** [review]

File: static/app/components/profiling/frameStack.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/components/profiling/frameStack.tsx:88]: I'll check, I need to cross verify what some tools do here, often times when you are looking at inverted, it's default sorted by self weight and when its call order, you'll get by total weight. I'll try setup some sensible defaults and look into removing the null

**Pull request merged**